### PR TITLE
feat: [FC-3] custom text input

### PR DESCRIPTION
**Pull request by huydosgtech**

### CODE REVIEW

**Comment by huydosgtech** [comment]
Demo:

Simulator Screenshot - iPhone 13 - 2024-06-06 at 11 11 30

File: src/chat/components/InputToolbar.tsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  customViewStyle?: StyleProp<ViewStyle>;
+  iconCamera?: any;
+  iconGallery?: any;
+  iconSend?: any;
+  iconStyle?: StyleProp<ImageStyle>;
+  iconSize?: number;
+  iconMargin?: number;
+}
+
+export const PressableIcon: React.FC<{
```
Comment (honghoangsts) [inline src/chat/components/InputToolbar.tsx:34]: move this implement to `PressableIcon.tsx`

File: src/chat/components/InputToolbar.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  iconStyle,
+  onPress,
+  resizeMode = 'contain',
+  disabled,
+  hitSlop,
+  size = 18,
+}) => (
+  <TouchableOpacity
+    disabled={disabled}
+    hitSlop={{ top: 10, right: 10, bottom: 10, left: 10, ...hitSlop }}
```
Comment (honghoangsts) [inline src/chat/components/InputToolbar.tsx:55]: create constant for hitSlop

File: src/chat/components/InputToolbar.tsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  textInput: {
+    marginHorizontal: 20,
+    lineHeight: 20,
+  },
+  marginWrapperView: {
+    marginRight: 10,
+  },
+});
+
+InputToolbar.defaultProps = {
```
Comment (honghoangsts) [inline src/chat/components/InputToolbar.tsx:151]: why we need this?

File: src/chat/components/InputToolbar.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+      )}
+      <View style={[styles.composeWrapper, composeWrapperStyle]}>
+        <ScrollView scrollEnabled={false}>
+          <Composer
+            {...props}
+            textInputStyle={[styles.textInput, composerTextInputStyle]}
+          />
+        </ScrollView>
+      </View>
+      {text ? (
```
Comment (honghoangsts) [inline src/chat/components/InputToolbar.tsx:112]: parse to boolean

File: src/chat/ChatScreen.tsx [hunk 5]
```diff
@@ -96,6 +102,14 @@ export const ChatScreen: React.FC<ChatScreenProps> = ({
     [firebaseInstance, memberIds, partners]
   );
 
+  const customInputToolbar = useCallback(
+    (composeProps: ComposerProps) => {
+      if (renderComposer) return renderComposer(composeProps);
+      return <InputToolbar {...composeProps} />;
```
Comment (honghoangsts) [inline src/chat/ChatScreen.tsx:108]: how to custom style for `InputToolbar`

File: src/chat/components/InputToolbar.tsx [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+const InputToolbar: React.FC<IInputToolbar> = ({
+  isShowCamera = true,
+  isShowPhotoGallery = true,
+  togglePhotoGallery,
+  containerStyle,
+  composeWrapperStyle,
+  composerTextInputStyle,
+  customViewStyle,
+  iconCamera = require('../../images/camera.png'),
```
Comment (honghoangsts) [inline src/chat/components/InputToolbar.tsx:75]: create constant for images

File: src/chat/components/InputToolbar.tsx [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+  iconGallery = require('../../images/gallery.png'),
+  iconSend = require('../../images/send.png'),
+  iconStyle,
+  iconSize = 28,
+  iconMargin = 12,
```
Comment (honghoangsts) [inline src/chat/components/InputToolbar.tsx:80]: change to iconStyle so user can have more custom

File: src/chat/components/InputToolbar.tsx [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+        <PressableIcon
+          style={{ marginHorizontal: iconMargin }}
+          size={iconSize}
+          icon={iconCamera}
+          iconStyle={iconStyle}
+        />
+      )}
+      {isShowPhotoGallery && (
+        <PressableIcon
+          onPress={togglePhotoGallery}
```
Comment (honghoangsts) [inline src/chat/components/InputToolbar.tsx:97]: where is implement for gallery?

File: src/chat/components/InputToolbar.tsx [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+  iconSize = 28,
+  iconMargin = 12,
+  ...props
+}) => {
+  const { onSend, text } = props;
+
+  return (
+    <View style={[styles.container, containerStyle]}>
+      {isShowCamera && (
+        <PressableIcon
```
Comment (honghoangsts) [inline src/chat/components/InputToolbar.tsx:88]: where is implement for show camera?

File: src/images/camera.png
Comment (honghoangsts) [inline src/images/camera.png:1]: use the image smaller

File: src/chat/components/InputToolbar.tsx [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+      {text ? (
+        <PressableIcon
+          style={{ marginHorizontal: iconMargin }}
+          onPress={() => onSend?.({ text: text }, true)}
+          size={iconSize}
+          icon={iconSend}
+          iconStyle={iconStyle}
+        />
+      ) : (
+        <View style={[styles.marginWrapperView, customViewStyle]} />
```
Comment (honghoangsts) [inline src/chat/components/InputToolbar.tsx:121]: why we need this view?

File: src/chat/components/InputToolbar.tsx (hunk 9 above)
Comment (huydosgtech) [reply src/chat/components/InputToolbar.tsx:56]: I updated, please check it

File: src/chat/components/InputToolbar.tsx (hunk 8 above)
Comment (huydosgtech) [reply src/chat/components/InputToolbar.tsx:63]: I updated, please check it

File: src/chat/components/InputToolbar.tsx (hunk 1 above)
Comment (huydosgtech) [reply src/chat/components/InputToolbar.tsx]: Yes, I moved,  please check it

File: src/chat/components/InputToolbar.tsx (hunk 2 above)
Comment (huydosgtech) [reply src/chat/components/InputToolbar.tsx]: Yes, I created a constant,  please check it

File: src/chat/components/InputToolbar.tsx (hunk 3 above)
Comment (huydosgtech) [reply src/chat/components/InputToolbar.tsx]: I removed it

File: src/chat/ChatScreen.tsx (hunk 5 above)
Comment (huydosgtech) [reply src/chat/ChatScreen.tsx]: I added styleInputToolbar to custom this component

File: src/chat/components/InputToolbar.tsx (hunk 6 above)
Comment (huydosgtech) [reply src/chat/components/InputToolbar.tsx]: yes, I created

File: src/chat/components/InputToolbar.tsx (hunk 7 above)
Comment (huydosgtech) [reply src/chat/components/InputToolbar.tsx]: yes, removed iconSize, iconMargin, only need use **iconStyle**

File: src/chat/components/InputToolbar.tsx (hunk 9 above)
Comment (loc-nguyenthien) [reply src/chat/components/InputToolbar.tsx:54]: I still haven't see `onPress` prop

File: src/chat/components/InputToolbar.tsx [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+        <ScrollView scrollEnabled={false}>
+          <Composer
+            {...props}
+            textInputStyle={[styles.textInput, composerTextInputStyle]}
+          />
+        </ScrollView>
+      </View>
+      {!!text && (
+        <PressableIcon
+          iconStyle={[styles.iconStyleDefault, iconStyle]}
```
Comment (loc-nguyenthien) [inline src/chat/components/InputToolbar.tsx:76]: I recommend use `StyleSheet.flatten([...])` for all style that we allow customize

File: src/chat/components/PressableIcon.tsx [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+};
+
+export const PressableIcon: React.FC<{
+  icon: any;
+  style?: StyleProp<ViewStyle>;
+  iconStyle?: StyleProp<ImageStyle>;
+  onPress?: () => void;
+  resizeMode?: 'contain' | 'cover' | 'stretch' | 'repeat' | 'center';
+  disabled?: boolean;
+  hitSlop?: { top?: number; right?: number; bottom?: number; left?: number };
```
Comment (loc-nguyenthien) [inline src/chat/components/PressableIcon.tsx:24]: `Insets` type from 'react-native'

File: src/chat/components/PressableIcon.tsx [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+  bottom: 10,
+  left: 10,
+};
+
+export const PressableIcon: React.FC<{
+  icon: any;
+  style?: StyleProp<ViewStyle>;
+  iconStyle?: StyleProp<ImageStyle>;
+  onPress?: () => void;
+  resizeMode?: 'contain' | 'cover' | 'stretch' | 'repeat' | 'center';
```
Comment (loc-nguyenthien) [inline src/chat/components/PressableIcon.tsx:22]: `ImageResizeMode` type from 'react-native'

**Review by loc-nguyenthien — CHANGES_REQUESTED** [review]

**Review by honghoangsts — CHANGES_REQUESTED** [review]

File: src/chat/components/InputToolbar.tsx (hunk 10 above)
Comment (huydosgtech) [reply src/chat/components/InputToolbar.tsx]: I removed it

File: src/chat/components/InputToolbar.tsx (hunk 11 above)
Comment (huydosgtech) [reply src/chat/components/InputToolbar.tsx:76]: ok I updated it

**Review by honghoangsts — APPROVED** [review]

**Pull request merged**